intake/catalog/utils.py

```diff
@@ -133,6 +174,39 @@ def expand_defaults(default, client=False, getenv=True, getshell=True):
     return default
 
 
+def merge_pars(params, user_inputs, spec_pars, client=False, getenv=True,
+               getshell=True):
```

Comment:
doc string?

Reply:
Yep, lots of detailed documentation to come...